### ISSUE TITLE
DIRAC's Epoch time has time zone dependence

### ISSUE THREAD

**Issue by hmiyake** [DIRACGrid/DIRAC#5275]
I realized Time.toEpoch() returns different values depending on time zone.

DIRAC/Core/Utilities/Time.py
>  def toEpoch(dateTimeObject=None):
>...
>   return nativetime.mktime(dateTimeObject.timetuple())

However, dateTime() returns
> return dt.utcnow()

I think toEpoch() should return
>  import calendar
>  return calendar.timegm(dateTimeObject.timetuple())

to be universal among multiple time zones

Similary fromEpoch() should be
>  return dt.utcfromtimestamp(epoch)

However, I wonder if we can apply this change without thinking carefully since timestamp will be "leaped" after the modification. Accounting plots will have a gap and duplication, and even contradicted DB records could be produced...
So first of all, I'd confirm if my thought is correct, then want to consult how to address the issue.

**Comment by fstagni** [comment]
Indeed, `toEpoch` it's using the local time. Practically speaking, what are the visible consequences right now?

**Comment by hmiyake** [comment]
We have major DIRAC nodes at US and Japan. US node always fail to register the activity due to "invalid for activity" in Framework/Monitoring.commitMarks()
We also faced timezone issue at our ES evaluation nodes (Kibana and DIRAC showed time difference). ElasticSearchDB.py makes "timestamp" from Time.toEpoch()...do you mean it is not UTC but local time? (recently I don't check the detail, so maybe I misunderstand anything about ES-Kibana configuration)

**Comment by fstagni** [comment]
```
In [1]: import time                                                                                                                                                                                                

In [2]: time.mktime?                                                                                                                                                                                               
Docstring:
mktime(tuple) -> floating point number

Convert a time tuple in local time to seconds since the Epoch.
Note that mktime(gmtime(0)) will not generally return zero for most
time zones; instead the returned value will either be equal to that
of the timezone or altzone attributes on the time module.
Type:      builtin_function_or_method
```

yes, it's local time :(

Framework/Monitoring is old and crappy/buggy so I would not put effort into it. But for accounting and ES/Monitoring looks like it can indeed be problematic.

**Comment by fstagni** [comment]
Brief summary after today's BiLD meeting: somehow we need to bite the bullet: we should move to UTC everywhere. This should only be done in a major release.
